lib/LWP/Simple.pm

```diff
@@ -87,17 +103,23 @@ method request_shell (RequestType $rt, Str $url, %headers = {}, Any $content?) {
     my ($status, $resp_headers, $resp_content) =
         self.make_request($rt, $hostname, $port, $path, %headers, $content, :$ssl);
 
-    # HTML header names can be mixed case. To find the desired names, transform
-    # them into lowercase.
-    my %resp_header_lowercase;
-    for $resp_headers.keys -> $header_name {
-        %resp_header_lowercase{$header_name.lc} = $resp_headers{$header_name};
-    }
-
     given $status {
 
+        when / <[4..5]> <[0..9]> <[0..9]> / {
```

Comment:
Wow, what a way to check `400 <= $_ <= 599`

Reply:
Which I guess either are substantially quicker than a regex.

I think if we merge this, I'll fix it in another commit as this was a pre-existing condition :-)